### PR TITLE
Pull Platform out of SysrootCreator for reuse

### PR DESCRIPTION
**Pull request by emersonknapp**
* DockerConfig and Platform had close knowledge of each other, consolidate their responsibility
* Moving Platform out into a separate file so that it can be used by other upcoming steps of the cross-compile pipeline.
* Removed the dangling unused field `cc_toolchain`

Signed-off-by: Emerson Knapp <emerson.b.knapp@gmail.com>

### CODE REVIEW

File: ros_cross_compile/platform.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+# limitations under the License.
+import getpass
+from typing import Optional
+
+# NOTE: when changing any following values, update README.md Supported Targets section
+ARCHITECTURE_DOCKER_MAP = {
+    'armhf': 'arm32v7',
+    'aarch64': 'arm64v8',
+}
+SUPPORTED_ARCHITECTURES = list(ARCHITECTURE_DOCKER_MAP.keys())
```
Comment (thomas-moulard) [inline ros_cross_compile/platform.py]: Consider using a frozenset instead of a list as this is immutable.

File: ros_cross_compile/platform.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+SUPPORTED_ROS2_DISTROS = ['dashing', 'eloquent']
+SUPPORTED_ROS_DISTROS = ['kinetic', 'melodic']
```
Comment (thomas-moulard) [inline ros_cross_compile/platform.py]: ditto (frozenset)

File: ros_cross_compile/platform.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        if self.ros_distro in SUPPORTED_ROS2_DISTROS:
+            self._ros_version = 'ros2'
+        elif self.ros_distro in SUPPORTED_ROS_DISTROS:
+            self._ros_version = 'ros'
+        else:
+            raise ValueError('Unknown ROS distribution "{}" specified'.format(ros_distro))
+
+        if self.os_name not in ROSDISTRO_OS_MAP[self.ros_distro]:
+            raise ValueError(
+                'OS "{}" not supported for ROS distro "{}"'.format(os_name, ros_distro))
```
Comment (thomas-moulard) [inline ros_cross_compile/platform.py]: Whenever possible, I'd rather let advanced users do whatever they want to try things out easily. Could we change that to a warning, or an error and let users figure out by themselves? We could always say there is no QA/support for this configuration.

Comment (emersonknapp) [reply ros_cross_compile/platform.py]: I don't specifically disagree, but changing this functionality is out of scope for this PR, which maintains current behavior

**Comment by emersonknapp** [comment]
I made the types immutable via tuple instead of frozenset

File: ros_cross_compile/platform.py (hunk 3 above)
Comment (thomas-moulard) [reply ros_cross_compile/platform.py]: sgtm!

**Pull request merged**